tests/handlers/test_profile.py

```diff
@@ -327,6 +327,47 @@ def test_avatar_constraint_mime_type(self) -> None:
         )
         self.assertFalse(res)
 
+    @unittest.override_config({"allowed_avatar_mimetypes": ["image/png"]})
+    def test_avatar_fetching_metadata_right_source(self) -> None:
+        """Tests that local and remote file are rightly fetched for metadata
+        when checking for avatar size and mime type"""
+        remote_server = "test:8080"
+        store = self.hs.get_datastores().main
+
+        self.get_success(
+            store.store_local_media(
+                media_id="local",
+                media_type="image/png",
+                media_length=50,
+                time_now_ms=self.clock.time_msec(),
+                upload_name=None,
+                user_id=UserID.from_string("@whatever:test"),
+            )
+        )
+        self.get_success(
+            store.store_cached_remote_media(
+                media_id="remote",
+                media_type="image/png",
+                media_length=50,
+                origin=remote_server,
+                time_now_ms=self.clock.time_msec(),
+                upload_name=None,
+                filesystem_id="remote",
+            )
+        )
+
+        res = self.get_success(
+            self.handler.check_avatar_size_and_mime_type("mxc://test/local")
+        )
+        self.assertTrue(res)
+
+        res = self.get_success(
+            self.handler.check_avatar_size_and_mime_type(
+                "mxc://" + remote_server + "/remote"
+            )
+        )
```

Comment:
It would be good to add some comments about what exactly this is testing, and how. It looks like we are not attempting to simulate a fetch from a remote server? Is the idea that, if the media is cached, we shouldn't attempt to fetch it?

Reply:
This should address all your remarks:

The issue I discovered was when non-default port is used for server, `check_avatar_size_and_mime_type` attempts to get metadata from wrong source - remote media cache instead of local.

Hence this test ensures function is looking at the right place. It doesn't care for whether `get_local_media()` or `get_cached_remote_media()` since that would be best handled by the tests written to test their output.

This is why I had the name of test as `test_avatar_fetching_metadata_right_source`. If this sounds good to you, I can add the same as comments in code.

Reply:
yes, please add some comments to the code

Reply:
Added!